packages/emitter-framework/src/python/components/enum-declaration/enum-declaration.test.tsx

```diff
@@ -0,0 +1,246 @@
+
+
+    `);
+  });
+
+  it("can be referenced", async () => {
+    const result = await runner.compile(t.code`
+      enum ${t.enum("Foo")} {
+        one: 1,
+        two: 2,
+        three: 3
+      }
+    `);
+
+    const output = getOutput(runner.program, [
+      <EnumDeclaration type={result.Foo} />,
+      <StatementList>
+        {efRefkey(result.Foo)}
+        {efRefkey(result.Foo.members.get("one"))}
+      </StatementList>,
+    ]);
+
+    expect(output).toRenderTo(d`
+      from enum import IntEnum
+
+      class Foo(IntEnum):
+        ONE = 1
+        TWO = 2
+        THREE = 3
+
+
+      Foo;
+      Foo.ONE;
+    `);
+  });
+
+  it("can be referenced using union", async () => {
+    const result = await runner.compile(t.code`
+      union ${t.union("Foo")}  {
+        one: 1,
+        two: 2,
+        three: 3
+      }
+    `);
+
+    const output = getOutput(runner.program, [
+      <EnumDeclaration type={result.Foo} />,
+      <StatementList>
+        {efRefkey(result.Foo)}
+        {efRefkey(result.Foo.variants.get("one"))}
+      </StatementList>,
+    ]);
+
+    expect(output).toRenderTo(d`
+      from enum import IntEnum
+
+      class Foo(IntEnum):
+        ONE = 1
+        TWO = 2
+        THREE = 3
+
+
+      Foo;
+      Foo.ONE;
+    `);
+  });
+
+  describe("Enum Type Detection", () => {
+    it("generates IntEnum for integer values", async () => {
+      const result = await runner.compile(t.code`
+        enum ${t.enum("StatusCode")} {
+          success: 200,
+          notFound: 404,
+          serverError: 500
+        }
+      `);
+      const output = getOutput(runner.program, [<EnumDeclaration type={result.StatusCode} />]);
+
+      expect(output).toRenderTo(d`
+        from enum import IntEnum
+
+        class StatusCode(IntEnum):
+          SUCCESS = 200
+          NOT_FOUND = 404
+          SERVER_ERROR = 500
+
+
+      `);
+    });
+
+    it("generates StrEnum for string values", async () => {
+      const result = await runner.compile(t.code`
+        enum ${t.enum("Color")} {
+          red: "red",
+          green: "green",
+          blue: "blue"
+        }
+      `);
+      const output = getOutput(runner.program, [<EnumDeclaration type={result.Color} />]);
+
+      expect(output).toRenderTo(d`
+        from enum import StrEnum
+
+        class Color(StrEnum):
+          RED = "red"
+          GREEN = "green"
+          BLUE = "blue"
+
+
+      `);
+    });
+
+    it("generates Enum for mixed value types", async () => {
+      const result = await runner.compile(t.code`
+        enum ${t.enum("Mixed")} {
+          stringValue: "hello",
+          numericValue: 42,
+          defaultValue
+        }
+      `);
+      const output = getOutput(runner.program, [<EnumDeclaration type={result.Mixed} />]);
+
+      expect(output).toRenderTo(d`
+        from enum import Enum
+
+        class Mixed(Enum):
+          STRING_VALUE = "hello"
+          NUMERIC_VALUE = 42
+          DEFAULT_VALUE = "defaultValue"
```

Comment:
If there is no value, I don't think we should assume that the value should be the name, but instead use [`enum.auto()`](https://docs.python.org/3/library/enum.html#enum.auto)